### DISCUSSION TITLE
Add URL parameter in API client

### DISCUSSION THREAD

**Discussion by oleksiivelychko** [go-swagger/go-swagger#2742] [General]
I got generated API client and I've started write tests for it. But I came across to the next problem - I didn't find how can I pass URL in:path parameter into my API client.

```
       params := products.NewUpdateProductParams()

	var pName = "Morning coffee"
	params.Body = &models.Product{
		Name: &pName,
	}

	_, err := client.Products.UpdateProduct(params)
	if err != nil {
		t.Fatal(err)
	}
```

In this above code I want to specify ID parameter for the exists route _/prodcuts/{id}_
How can I do it?

**Comment by casualjim** [comment]
Normally there should be  a property ID on the params struct.

```go
params := products.NewUpdateProductParams()
params.ID = 123456
```

**Reply by oleksiivelychko** [reply]
If I assign ID to params variable then I will get `Unresolved reference 'ID'` because UpdateProductParams struct doesn't have that field.

```
type UpdateProductParams struct {

	/* Body.

	Product data structure to Update or Create.
	Note: the id field is ignored by update and create operations
	*/
	Body *models.Product

	timeout    time.Duration
	Context    context.Context
	HTTPClient *http.Client
}
```

**Reply by casualjim** [reply]
Why would it be a path param but not on the params struct. Is there a path param in the spec?

Would it be possible to the operation from the spec?

**Reply by oleksiivelychko** [reply]
Yes, you right. I missed ID param in the swagger annotation.
Thanks for support.
